typed-racket-lib/typed-racket/env/lexical-env.rkt

```diff
@@ -95,6 +99,15 @@
                                                  Err))
                                    (register-type i t)
                                    t)]
+                             [(and (let* ([fail (λ () #f)]
+                                          [v (syntax-local-value i fail)]
+                                          [orig-id (and (provide/contract-info? v)
+                                                        (provide/contract-info-original-id v))])
+                                     (and (identifier? orig-id)
+                                          (lookup-type orig-id fail))))
+                              => (λ (orig-t)
+                                   (register-type i orig-t)
+                                   orig-t)]
```

Comment:
@samth this is the part of Shallow TR that goes behind contracts to get types.

Ideally we'd use the `typecheck/renamer.rkt` interface to get types, but (I believe) the typechecking code here gets run after expansion => after those rename transformers have been expanded to the contracted identifiers.

Another idea is to serialize more types for the new identifiers, but I was never able to make that work. The typed id's come from one submodule & the contract versions come from another ... and I couldn't find a meeting point.